From: Sang-Heon Jeon <ekffu200098@gmail.com>
Date: Fri, 15 Aug 2025 18:40:59 +0900
Subject: [PATCH] mm/damon/core: fix damos_commit_filter not changing allow

Current damos_commit_filter() not persist allow value of filter. As a
result, changing allow value of filter and commit doesn't change
allow value.

Add the missing allow value update, so commit filter now persist changing
allow value well.

Fixes: fe6d7fdd6249 ("mm/damon/core: add damos_filter->allow field")
Cc: stable@vger.kernel.org # 6.14.x
Signed-off-by: Sang-Heon Jeon <ekffu200098@gmail.com>
Link: https://patch.msgid.link/20250815094059.133769-1-ekffu200098@gmail.com
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/core.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/damon/core.c b/mm/damon/core.c
index 19c8f01fc81a..cb41fddca78c 100644
--- a/mm/damon/core.c
+++ b/mm/damon/core.c
@@ -900,6 +900,7 @@ static void damos_commit_filter(
 {
 	dst->type = src->type;
 	dst->matching = src->matching;
+	dst->allow = src->allow;
 	damos_commit_filter_arg(dst, src);
 }
 
-- 
2.39.5

